### PR TITLE
Update Microsoft.PolicyInsights module

### PR DESCRIPTION
**Pull request by eladperets**
## Description

- Updated our SDK to its latest version
- Added support to the new properties in the remediation commands + made minor changes to paging logic to reflect the underlying API changes
- Update policy states and policy events commands to work with the new API version that now supports paging.
- Added\updated tests to validate changes

Testing paging (and testing this module in general) requires a complex environment with large number of resources. Created a script that will setup the test env from scratch to make things easier for future devs.
 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by eladperets** [comment]
Static analysis is failing because:
`Parameter ParallelDeployments of cmdlet Start-AzPolicyRemediation does not follow the enforced naming convention of using a singular noun for a parameter name.`
`ParallelDeployments` is the name of the property on the underlying API, I don't think we need\should change it to singular.

**Comment by wyunchi-ms** [comment]
As PowerShell suggest use singularity for parameters. You can find doc [here](https://github.com/Azure/azure-powershell/blob/main/documentation/development-docs/design-guidelines/parameter-best-practices.md#singularity). Please rename `ParallelDeployment` to `ParallelDeployment`.

**Comment by eladperets** [comment]
> suggest

Hi @wyunchi-ms, As I mentioned above, the name of this parameter is identical the name of the corresponding resource property in the API. Using a singular form will be inconsistent with the API plus the name itself won't make any sense- the parameter determines the how many parallel deployments to run at any given time, changing it to singular will be very confusing.

**Comment by wyunchi-ms** [comment]
Hi @eladperets most of our customers are expert. They know the name rule of PowerShell. `Plural parameter` will break the consistent of our project and make them confused.

**Comment by eladperets** [comment]
> Hi @eladperets most of our customers are expert. They know the name rule of PowerShell. `Plural parameter` will break the consistent of our project and make them confused.

@wyunchi-ms , I totally understand the need for consistency, but the phrase `ParallelDeployment` doesn't make any sense in the context of what the operation is doing. Even without the context it's meaningless IMO- the word `Parallel` implies more than one operation. I can see many similar examples in other commands that were exempt from this rule, I think this case justify an exemption as well.

If you still disagree, is there some sort of PS design forum that can help with deciding what the best thing to do here?

File: src/PolicyInsights/PolicyInsights/Az.PolicyInsights.psd1 [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '1.4.1'
+ModuleVersion = '1.5.0'
```
Comment (wyunchi-ms) [inline src/PolicyInsights/PolicyInsights/Az.PolicyInsights.psd1]: Please don't update the version and release notes in this file manully, our tool will update them.

File: src/PolicyInsights/PolicyInsights/Cmdlets/Remediation/StartAzureRmPolicyRemediation.cs [hunk 2]
```diff
@@ -69,6 +69,18 @@ public class StartAzureRmPolicyRemediation : RemediationCmdletBase
         [ValidateSet("ExistingNonCompliant", "ReEvaluateCompliance")]
         public string ResourceDiscoveryMode { get; set; }
 
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.RemediationResourceCount)]
+        [ValidateRange(1, int.MaxValue)]
+        public int? ResourceCount { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = ParameterHelpMessages.RemediationParallelDeployments)]
+        [ValidateRange(1, int.MaxValue)]
+        public int? ParallelDeployments { get; set; }
```
Comment (wyunchi-ms) [inline src/PolicyInsights/PolicyInsights/Cmdlets/Remediation/StartAzureRmPolicyRemediation.cs]: Is it a better name `ParallelDeployCount`?

File: src/PolicyInsights/PolicyInsights/Az.PolicyInsights.psd1 (hunk 1 above)
Comment (wyunchi-ms) [reply src/PolicyInsights/PolicyInsights/Az.PolicyInsights.psd1]: And please update the changelog.md

File: src/PolicyInsights/PolicyInsights/Cmdlets/Remediation/StartAzureRmPolicyRemediation.cs (hunk 2 above)
Comment (eladperets) [reply src/PolicyInsights/PolicyInsights/Cmdlets/Remediation/StartAzureRmPolicyRemediation.cs:78]: There are 1116 existing exceptions for this rule across the Azure Powershell repo, many of them are very similar to this case. Why can't we have an exception here as well?

Comment (wyunchi-ms) [reply src/PolicyInsights/PolicyInsights/Cmdlets/Remediation/StartAzureRmPolicyRemediation.cs:78]: Please email to our team channel if you have any question.

Comment (eladperets) [reply src/PolicyInsights/PolicyInsights/Cmdlets/Remediation/StartAzureRmPolicyRemediation.cs:78]: @wyunchi-ms what is your team channel? couldn't find it in any public\internal docs.

Comment (eladperets) [reply src/PolicyInsights/PolicyInsights/Cmdlets/Remediation/StartAzureRmPolicyRemediation.cs:78]: @wyunchi-ms nvm. Changed it to `ParallelDeploymentCount`.

**Comment by eladperets** [comment]
> /azp run

@wyunchi-ms I also tried to run the pipelines, but I don't have permissions. Do you know why some checks timed out? Is that a blocker for merging the PR? I want to get it published as soon as possible.

File: src/PolicyInsights/PolicyInsights/ChangeLog.md [hunk 3]
```diff
@@ -19,6 +19,10 @@
 -->
 ## Upcoming Release
 
+## Version 1.5.0
```
Comment (wyunchi-ms) [inline src/PolicyInsights/PolicyInsights/ChangeLog.md]: Our tool will update the version. Please don't update it manully.

**Comment by wyunchi-ms** [comment]
Please don't modify our test framework to make test case pass.

**Review by wyunchi-ms — APPROVED** [review]

**Pull request merged**